topic-operator/src/main/java/io/strimzi/operator/topic/cruisecontrol/CruiseControlClientImpl.java

```diff
@@ -191,23 +195,28 @@ public UserTasksResponse userTasks(Set<String> userTaskIds) {
             }
         }).join();
     }
-    
+
     public Optional<String> errorMessage(HttpResponse<String> response) {
         if (response != null && response.body() != null && !response.body().isBlank()) {
-            try {
-                ErrorResponse errorResponse = objectMapper.readValue(response.body(), ErrorResponse.class);
-                if (errorResponse.errorMessage() != null) {
-                    if (errorResponse.errorMessage().contains("NotEnoughValidWindowsException")) {
-                        return Optional.of("Cluster model not ready");
-                    } else if (errorResponse.errorMessage().contains("OngoingExecutionException")
+            // for some reason CC returns an HTML page on authorization failures
+            if (response.body().contains("Unauthorized")) {
```

Comment:
Doesn't it have the code? In the mock, you seem to have the right status code. That suggests that the mock is wrong or this is not needed?

Reply:
Correct, I can use the code. The mock is still correct. I'm just using a simplified html compared to what's returned by the CC server.

Reply:
Yeah, I do not have problem with the body in the mock. The concern was more about the status code in the mock if it isn't returned by Cruise Control. But I think the mock is fine.